specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2024-05-15-preview/networkSecurityPerimeter.json

```diff
@@ -0,0 +1,455 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/networkSecurityPerimeterConfigurationName"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Request to reconcile the association accepted.",
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "NetworkSecurityPerimeterConfiguration": {
+      "description": "The Network Security Perimeter configuration resource.",
+      "type": "object",
+      "readOnly": true,
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/NetworkSecurityPerimeterConfigurationProperties",
+          "description": "Properties of the Network Security Perimeter Configuration",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationList": {
+      "description": "Result of the List Network Security Perimeter configuration operation.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "A collection of Network Security Perimeter configurations",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkSecurityPerimeterConfiguration"
+          }
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationProperties": {
+      "description": "Properties of the Network Security Perimeter Configuration",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of Network Security Perimeter configuration propagation",
+          "enum": [
+            "Accepted",
+            "Succeeded",
+            "Failed",
+            "Deleting",
+            "Canceled"
+          ],
```

Comment:
I see you don't define Creating. But BotService does. I wonder why. Is this actually up to each RP to decide their own set?

```
          "enum": [
            "Creating",
            "Updating",
            "Accepted",
            "Succeeded",
            "Failed",
            "Deleting"
          ],
```

Reply:
Added

Reply:
This looks like a good set which should serve most RP's needs.